code/modules/flufftext/Hallucination.dm

```diff
@@ -16,46 +16,66 @@ Gunshots/explosions/opening doors/less rare audio (done)
 	var/image/halbody
 	var/obj/halitem
 	var/hal_screwyhud = SCREWYHUD_NONE
-	var/handling_hal = 0
+	var/next_hallucination = 0
 
 /mob/living/carbon/proc/handle_hallucinations()
-	if(handling_hal)
+	if(world.time < next_hallucination)
 		return
-
 	//Least obvious
-	var/list/minor = list("sounds"=25,"bolts_minor"=5,"whispers"=15,"message"=10,"hudscrew"=15)
+	var/list/minor = list(
```

Comment:
All these lists should be static so they are not generated every time. Also name them something like `hallucinations_minor/medium/major`.